.editorconfig

```diff
@@ -1,16 +1,79 @@
 root = true
 
-[*]
+
+## Repository Configurations
+############################
+[.{git*,editorconfig,*.yml}]
 charset = utf-8
 end_of_line = lf
 indent_style = space
+indent_size = unset
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[.gitmodules]
+indent_style = tab
+
+
+## Shell Scripts
+################
+[*.sh]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
+indent_size = unset
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+
+## Make Files
+#############
+[Makefile]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
 indent_size = 2
+trim_trailing_whitespace = true
 insert_final_newline = true
+
+
+## Haskell Sources
+##################
+[*.hs]
+charset = utf-8
+end_of_line = unset
+indent_style = space
+indent_size = unset
```

Comment:
Yes it would fail because the validator tool expects all indentation to be a multiple of the base value. Unfortunately the EditorConfig project doesn't offer separate settings for editor/IDEs configurations and validators right now. 

Since most coders add extra spaces to obtain elegant alignments on code that spans multiple lines, it often becomes necessary to either unset `indent_size` or set it to `1` — the former is better because it doesn't override the editor's defaults, whereas the latter would actually enforce 1-space tabs.

As for markdown (and other lightweight markup syntaxes) the problem affect code blocks, which could be using even or odd indentation, depending on the language snippet being included.

Hopefully in future editions of EditorConfig we'll see new settings to address validation aspects only. Right now, one is forced to compromise on some features to obtain both editor settings and CI validation — e.g. EditorConfig doesn't (and will never) support `end_of_line = native`, so to allow cross-platform EOL normalization `end_of_line` has to be unset and the project has to rely on `.gitattributes` instead.

I'll take a further look at EClint's documentation, there are some ways to customize validation settings and overriding the `.editorconfig` values via a separate config file for EClint; if this is the case, we could enforce 2-spaces on the editor and override it on EClint by unsetting `indent_size` for some file types. It might require some time for me to dig into it and test it, but in the meantime we could use these settings for they are fairly good in practical use, from my own experience (you get a sound validation and don't interfere with the editor/IDE).

Reply:
I've checked EClint documentation, and there's no easy way to override `.editorconfig` settings on a per-file-type manner (you can override options globally for each run). The only alternative would be to run multiple EClint checks, each targetting a specific file extension, but it seems an overkill solution.